internal/migrations/backfill-digest/migrate_actions.sh

```
@@ -0,0 +1,100 @@
+    if [[ ! -f "$file" ]]; then
+        echo "❌ Error: Migration file not found: $file"
+        exit 1
+    fi
+
+    echo "📄 Running $file"
+
+    # Execute migration and check exit status
+    if ! kwil-cli exec-sql --file "$file" --private-key "$PRIVATE_KEY" --provider "$PROVIDER" --sync; then
+        echo "❌ Migration failed: $file"
+        echo "   Command: kwil-cli exec-sql --file \"$file\" --private-key \"***\" --provider \"$PROVIDER\" --sync"
+        echo "   Exit code: $?"
```

Shellcheck (0.10.0)

[warning] This $? refers to echo/printf, not a previous command. Assign to variable to avoid it being overwritten.

(SC2320)